goal_src/jak1/pc/progress-pc.gc

```diff
@@ -1777,6 +1769,10 @@
               (option-y y-off)
               (locked? #f)
               )
+          ;; Check if the option should be disabled
+          (when (and (!= (-> options index option-disabled-func) 0)
+                     (= ((-> options index option-disabled-func)) #t))
```

Comment:
I'm not clear on what you're suggesting.

Are you saying I can just do:
```lisp
(when (and (!= (-> options index option-disabled-func) 0)
               ((-> options index option-disabled-func)))
```
instead?

The idea was for the lambda added to the option to explicitly return a simple `true` or `false`, so that's why I check against `#t` explicitly.

Reply:
Yes, that is what I'm suggesting.

There's no benefit to checking `#t` directly for when you want a true or false response, and it's also non-standard. Another example where it would break unexpectedly is with using `or` and/or `and` (lol):
```lisp
(or *target* *sidekick*)
(and *target* *sidekick*)
```
`or` will never return `#t` here. `or` returns the value of the first truthy operand, or `#f` if all are not truthy.
`and` will also never return `#t`. similar to `or`, it returns the value of the last operand, or `#f` if any operand is not truthy.

see https://opengoal.dev/docs/reference/syntax/#truth